helm/staking-ledger-cron/mainnet-dump-staking-ledger-cronjob.yaml

```diff
@@ -28,5 +28,5 @@ spec:
               mountPath: "/gcloud/"
             # command: ["/usr/bin/dumb-init", '/cron_job_dump_ledger.sh']
-            command: ["/bin/bash", "-c", 'mina daemon --generate-genesis-proof true --peer-list-url https://storage.googleapis.com/mina-seed-lists/mainnet_seeds.txt --background;sleep 480;echo "done sleeping";while true;do mina ledger export staking-epoch-ledger>staking_epoch_ledger.txt;if [ "$?" -eq 0 ]&&[ "$(cat staking_epoch_ledger.txt)" != "Ledger not found: current staking ledger not available" ];then echo "staking epoch ledger dumped!";break;else echo "waiting for staking ledger to become available, sleeping for 30s";sleep 30;fi;done;DATE="$(date +%F_%H%M)";STAKING_LEDGER_HASH="$(mina ledger hash --ledger-file staking_epoch_ledger.txt)";LEDGER_FILENAME="$DATE"_staking_epoch_ledger_"$STAKING_LEDGER_HASH".txt;mv ./staking_epoch_ledger.txt ./$LEDGER_FILENAME;mina ledger export next-epoch-ledger>next_epoch_ledger.txt;echo "next epoch ledger dumped!";NEXT_LEDGER_HASH="$(mina ledger hash --ledger-file next_epoch_ledger.txt)";NEXT_LEDGER_FILENAME="$DATE"_next_epoch_ledger_"$NEXT_LEDGER_HASH".txt;mv ./next_epoch_ledger.txt ./$NEXT_LEDGER_FILENAME;echo "upload to a GCP cloud storage bucket";gsutil -o Credentials:gs_service_key_file=/gcloud/keyfile.json cp $LEDGER_FILENAME gs://mina-staking-ledgers;gsutil -o Credentials:gs_service_key_file=/gcloud/keyfile.json cp $NEXT_LEDGER_FILENAME gs://mina-staking-ledgers' ]
+            command: ["/bin/bash", "-c", 'mina daemon --generate-genesis-proof true --peer-list-url https://storage.googleapis.com/mina-seed-lists/mainnet_seeds.txt --background;sleep 480;echo "done sleeping";while true;do mina ledger export staking-epoch-ledger>staking_epoch_ledger.json;if [ "$?" -eq 0 ]&&[ "$(cat staking_epoch_ledger.json)" != "Ledger not found: current staking ledger not available" ];then echo "staking epoch ledger dumped!";break;else echo "waiting for staking ledger to become available, sleeping for 30s";sleep 30;fi;done;mina ledger export next-epoch-ledger>next_epoch_ledger.json;echo "next epoch ledger dumped!";EPOCHNUM="$(mina client status|grep "Best tip consensus time"|grep -o "epoch=[0-9]*"|sed "s/[^0-9]*//g")";STAKING_HASH="$(mina ledger hash --ledger-file staking_epoch_ledger.json)";STAKING_MD5="$(md5sum staking_epoch_ledger.json|cut -d " " -f 1)";LEDGER_FILENAME=staking-"$EPOCHNUM"-"$STAKING_HASH"-"$STAKING_MD5".json;mv ./staking_epoch_ledger.json ./$LEDGER_FILENAME;NEXT_STAKING_HASH="$(mina ledger hash --ledger-file next_epoch_ledger.json)";NEXT_STAKING_MD5="$(md5sum next_epoch_ledger.json|cut -d " " -f 1)";NEXT_FILENAME=next-staking-"$EPOCHNUM"-"$NEXT_STAKING_HASH"-"$NEXT_STAKING_MD5".json;mv ./next_epoch_ledger.json ./$NEXT_FILENAME;echo "upload to a GCP cloud storage bucket";gsutil -o Credentials:gs_service_key_file=/gcloud/keyfile.json cp $LEDGER_FILENAME gs://mina-staking-ledgers;gsutil -o Credentials:gs_service_key_file=/gcloud/keyfile.json cp $NEXT_FILENAME gs://mina-staking-ledgers' ]
```

Comment:
maybe we should make an issue to replace this block of bash code with the file? I suppose there was a problem with that before.

Anyway, that wasn't introduced in this PR so I don't think it should block landing this.

Reply:
i was using the block of bash code bc it was easier/much faster to iterate on, and the result is exactly the same.  yes ideally it's all packaged into a nice script that's already present on the image, but then we'd have to wait for this PR to land in compatible, then use the latest compatible image.  copy pasting the script means i don't have to worry if the image we're using has it baked in

we can make a new issue for this, sure.